### DISCUSSION TITLE
Is it possible to change the rect of a page after its creation?

### DISCUSSION THREAD

**Discussion by jloehel** [pymupdf/PyMuPDF#845] [Q&A] [answered]
How can I change the rect of a page?

~~~Python
(Pdb) newpage
page 2 of <new PDF, doc# 2>
(Pdb) page
{'rect': Rect(0.0, 0.0, 720.0, 540.0), 'image': <_io.BytesIO object at 0x7fe48694cf68>}
(Pdb) newpage.rect
Rect(0.0, 0.0, 595.0, 842.0)
(Pdb) newpage.rect = page["rect"]
*** AttributeError: can't set attribute
~~~

**Comment by JorjMcKie** [comment, marked as answer]
> How can I change the rect of a page?

Sure you can do that  ... at the risk of displacing stuff already on the page. But if the page is empty, simply do ``page.setMediaBox(rect)`` :

```python
>>> import fitz
>>> doc=fitz.open()
>>> page=doc.new_page()
>>> page.rect
Rect(0.0, 0.0, 595.0, 842.0)
>>> page.setMediaBox(fitz.Rect(fitz.PaperRect("a3")))
>>> page.rect
Rect(0.0, 0.0, 842.0, 1191.0)
>>> 
```

In addition, once you have inserted a few things of which you know which rectangle ``rect`` contains them all, you can "shrink" the **_visible_** page rectangle to that rect:
```python
rect = fitz.Rect(100, 100, 400, 500)  # assuming this contains the stuff that should be visible
page.setCropBoc(rect)
>>> page.rect
Rect(0.0, 0.0, 300.0, 400.0)
>>> # but the MediaBox is left unchanged!
>>> page.MediaBox
Rect(0.0, 0.0, 842.0, 1191.0)
>>> 
```
If you print the page or make a pixmap of it, then only the cropbox is shown ...

**Reply by JorjMcKie** [reply]
Additional comments:
Imagine MediaBox to be the "physical" page rectangle. In relation to this, all insertion coordinates are computed.
Changing this for a non-empty page will therefore reposition implicitly the content. But it is still possible to do that, technically.

The CropBox is the visible sub-rectangle of the MediaBox. If present at all and if there is no page rotation, its dimension is equal to `page.rect` - potentially not starting with its top-left at (0,0) (wich `page.rect` **always** does).
